hw/ip/i2c/doc/_index.md

```diff
@@ -1,3 +1,225 @@
+(Note: The rise time cannot be explicitly controlled by internal hardware, and will be a function of the capacitance of the bus.
+Thus this parameter is largely budgetary, meaning that it tells the state machine how much time to wait for an RC rise.)
+- T<sub>f</sub>: set in register {{< regref TIMING1.T_F >}}.
+(Note: The fall time cannot be explicitly controlled by internal hardware, and is a function of the pin driver.
+Thus this parameter is also budgetary.
+Given that the actual fall time cannot be controlled to stay above the minimum values set in Table 10 of the specification, and so this in this regard this module currently is not strictly compliant to the I2C spec.) 
+- T<sub>SU,STA</sub>: set in register {{< regref TIMING2.TSU_STA >}}
+- T<sub>HD,STA</sub>: set in register {{< regref TIMING2.THD_STA >}}
+- T<sub>SU,DAT</sub>: set in register {{< regref TIMING3.TSU_DAT >}}.
+Taken to be synonymous with T<sub>SU,ACK</sub> 
+- T<sub>HD,DAT</sub>: set in register {{< regref TIMING3.THD_DAT >}}.
+Taken to be synonymous with T<sub>HD,ACK</sub>.
+Moreover, since the pin driver fall time is likely to be less then one clock cycle, this parameter is also taken to be synonymous with the parameters T<sub>VD,DAT</sub> and T<sub>VD,ACK</sub>
+- T<sub>SU,STO</sub>: set in register {{< regref TIMING4.TSU_STO >}}.
+- T<sub>BUF</sub>: set in register {{< regref TIMING4.TSU_BUF >}}
+
+The values programmed into the registers {{< regref TIMING0 >}} through {{< regref TIMING4 >}} are to be expressed in units of the bus clock period.
+Note in order to ensure compliance with the I2C spec, firmware must program these registers with values within the ranges laid out in Table 10 of the specification.
+These values can be directly computed using DIFs given the desired speed standard, the desired operating frequency, and the actual line capacitance.
+These timings are then fed directly to the I2C state machine to control the bus timing.  
+
+### Timeout Control
+A malfunctioning (or otherwise very slow) target device can hold SCL low indefinitely stalling the bus.
+For this reason the {{< regref TIMEOUT_CTRL >}} provide a clock-stretching timeout mechanism to notify firmware of this sort of condition.
+If {{< regref TIMEOUT_CTRL.EN >}} is asserted, an interrupt will be asserted the IP detects that a remote device has been holding SCL low for more than {{< regref TIMEOUT_CTRL.VAL >}} clock ticks.
+
+
+This feature is added as a utility, though it is not required by the I2C specification.
+However, in some applications it could be used in protocols which build upon I2C.
+For instance, SMBus applications using this IP could in principle use this to support SMBus timeouts.
+(Note: This is just an example application of this feature.
+Other features may also be required for complete SMBus functionality.)
+
+
+### Timing Control Registers
+The I2C module has a few interrupts including general data flow interrupts and unexpected event interrupts.
```

Comment:
is it just me or does the title here not match the content? The titles seems to be about timing control, but the content is mostly describing the interrupts and the conditions where they arise.

Reply:
> is it just me or does the title here not match the content? The titles seems to be about timing control, but the content is mostly describing the interrupts and the conditions where they arise.

Huh.  You are right it doesn't.   I should have caught that.  Fixed.